### DISCUSSION TITLE
Request: Persistent High Alert as feature/checkbox within main Alerts/Alert List + Alert profiles

### DISCUSSION THREAD

**Discussion by Teute0815** [NightscoutFoundation/xDrip#2091] [Ideas]
I'm a late eater/sleeper. So my glucose is not quite stable going into the night/sleeping (I know it's not a good idea).

Even though calculations might be right, my alarms go off while sleeping, because they cross the threshold (sometimes several times) before I'm headed the right direction.

This is keeping me from sleeping and many times I don't even know whether really to inject or wait 10 - 20 Minutes (which is okay/doable at daytime, but not at night).

For that "problem" I'd love a feature like option "persistent high alert" (settings/alarms & alerts/extra alerts (sDrip+) within the settings for "Glucose Level Alerts"

The actual option seems to be a bit to rudimentary/strict for me.
No timeplan possible
No individual threshold (high/low fetched from "high value" <- which one? The standard setting (settings/glucose units) or from the one from current timeplan at that moment?

My individual example:
I'd love to have a persitent high alert, if I'm above 10 mmol for 60 minutes from 01 a.m. to 08 a.m.

If that is working, there's actually an extra request:
Several "profiles in "glucose alert list"
So I could have a "standard day" and a "workshop day" profile

For working in front of people and not wanting to have an alarm straight away after coffee break, cookies etc., but a reminder after 60 minutes or so (persistent high alert), if action was correct/forgotten.
As workshops are not a day to day thing, it would be cool to have an extra "workshop day" profile, which could be easily chosen just in the morning by switching profiles from standard to workshop. A "current profile" notification/button should/could be displayed on startscreen, to easily be aware and switch again.

I hope, I've been not to stupid, just not to find/understand these already implemented fetures and some people agree.

Thanks

**Comment by Navid200** [comment]
The existing alerts are not consistent:
https://github.com/NightscoutFoundation/xDrip/issues/1762

I have missed the persistent high alert because it only sounds once.  I wish I could assign the ascending volume profile to it.  Same with the missed readings alert.

The problem you describe not wanting to get an alert in front of others, in my opinion, can be addressed by the ascending volume profile as it starts at a very low volume.

**Reply by Teute0815** [reply]
Thanks for your reply. For the workshop setting I've got different "workarounds" atm. The "alert profiles" request really is a bonus wish, once persistent high would be implemented into "Glucose Level Alerts" somehow.

Of much more importance to me would be the scheduled persistent high alert during night. That one is not solvable with ascending volume. I really just want to sleep until I've been at least 60min above threshold ;)

**Reply by Teute0815** [reply]
BTW. Do you know, which "high" value the persitant alert pulls? Is that the one from "glucose units"? Because those values (low/high) don't seem to have much more use than for graphic reasons on mainscreen - which I like to have "green" in between 4.5 and 8.5. So, I wouldn't like to set that on 10, just for the persitent alert, which probably easily could have an individually customizable threshold.

**Reply by Navid200** [reply]
> Of much more importance to me would be the scheduled persistent high alert during night. That one is not solvable with ascending volume. I really just want to sleep until I've been at least 60min above threshold ;)

You can set "For longer than" to 60 minutes.  Then, it will go off after you have been higher than your High level for more than 60 minutes.  
Can you explain why it's not solvable now?

The reference value for the persistent high alert is the High Value under Glucose Units setting.
Similarly, the reference value for the forecast low alert is the Low Value on the same page.

There will be no end in customization if we don't draw the line somewhere.  So, please explain why you need a different value.

**Reply by Teute0815** [reply]
**Ascending volume** would be of no help during night as my problem is not about the noise/disruption itself - it's about the fact, that I want to be (as) sure (as possible) that I'm too high (to my liking) and have been for a while (persistent).

Atm I do the workaround of snoozing high alerts for an hour before I go to sleep. But this still wakes me up later at night far to often for a short peak above threshold (but soon to would've been beneath again), rather than a "real too high and going/staying up"

**different threshold**: I (and my partner) do want to have as much consistent sleep as possible. So I'm willing to accept a higher threshold (10 mmol) thus more sleeping potential than my threshold at daytime which is set at 8.5 under "units" - mainly for graphic reasons but this is also my general "acceptable" high during daytime (as it is easier & less disturbing to act/react). My first "high alert" in "glucose levels" is actually set at 7.5 to alert me early enough, to be able to stay within 4.5-8.5 (at daytime - those alarms are excluded through timeplan during night.)

I really don't have a clue about programming part of things, but I thought/my hope was that a checkbox "persistent high" for "x minutes" could be relatively easy implemented into the existing main "glucose alerts" where everything else (timeplan, individual ringtone, individuall threshold, snooze, ascending volume as wished by you, etc.) is already available/implemented and working very well. 

That way I could just add another "glucose level alert" to the list, with persistent high checked for a specific time of day with individual threshold, which doesn't interfere with the graphics on mainscreen or other alerts/settings, which might pull the value from "glucose units"

The idea of a possibility to be able to switch between different "alert profiles" depending on current "way of life/work" was just a bonus. I've thought about that every now and then, when altering my list for hiking holidays or disabling alerts while in workshops and so on. You always have to remember to switch back and what the values actually were. Different profiles would make that much easier and with the "persistent high" maybe implemented, I thouhgt I'd just express that wish as well as it would be "just the next great step" then.

I'm not english native, so I'm not sure if I'm getting everything across properly and in a kind matter (which I really want). Please be asured, I'm not pressing anything and am really grateful for all the work all of you guys are doing.

I'll add a visual for better understanding:
Just a reminder: Even though the profiles are on top of the page (for better reference/understanding) - the pressing/main request is the checkbox "persistent" further down.
![2022-04-26 15 40 15](https://user-images.githubusercontent.com/61195783/165313449-7f16197e-5acc-45ad-9ed7-7dd9436690a7.jpg)

**Reply by Lupum1001** [reply]
> BTW. Do you know, which "high" value the persitant alert pulls? Is that the one from "glucose units"? Because those values (low/high) don't seem to have much more use than for graphic reasons on mainscreen - which I like to have "green" in between 4.5 and 8.5. So, I wouldn't like to set that on 10, just for the persitent alert, which probably easily could have an individually customizable threshold.

I think it uses those values set within "glucose units", yes.
I suspect those values are also used for "time in range" statistics though.

**Comment by Teute0815** [comment]
@Navid200
Thx for implementing thresholds for persistent hight alerts #3704 #3752 

Here's my old request, where you might want to read on why I would wish for a timeplan as well. If that is doable through other means already, I'm happy to learn about it.

**Reply by Navid200** [reply]
I have been concerned about persistent high alert and forecast low alert and missed reading alert not respecting volume profile and many other features being absent for them:
https://github.com/NightscoutFoundation/xDrip/discussions/3697 
  
Adding a threshold setting to the persistent high alert is a small step in the path of providing more control to users.  
  
I don't believe in the time of day or day of the week setting.  My diabetes does not understand a schedule.  I need to understand that I am diabetic, always every day every minute.  My diabetes does not go away when I have a meeting.  
Adding a time of day to an alert creates the possibility of the user forgetting that they customized an alert disabling it for a specific period of the day and finding out after it misbehaves.  

Before I go any further, please note that xDrip is only for research.  No one should use xDrip for controlling their diabetes.  A diabetic needs to see a doctor and follow the advice of their doctor including what app to use.  
All my posts in these forums are meant to help xDrip users use xDrip.  They are NOT meant to help users control their diabetes.

The purpose of the persistent high alert is for someone who is not on a low-carb diet.  I am not.  So, it is something I depend on.  When I go to sleep, if I go high, I don't want to get an alert as soon as I go high.  But, if it stays high and does not come back down, after a set time period, I would like to be notified.  That's what the persistent high alert does for me.

What I don't like about the persistent high alert is that its volume is controlled by the phone volume.  So, if I lower my phone volume while listening to a loud clip and forget to raise it back, it may never wake me up.
That's why I would like to add the feature to allow the persistent high alert to use the chosen volume profile.

**Reply by Teute0815** [reply]
Thanks for your reply.
>Adding a time of day to an alert creates the possibility of the user forgetting that they customized an alert disabling it for a specific period of the day and finding out after it misbehaves.

Isn't that contradicted by the fact, that xDrip offers the time plans for glucose alerts anyways? And many people use them, as they are very helpful. I'm quite sure, you do yourself (straight high alert at daytime/persistent at night - well, all day as it has no timeplan) The potentially heavily individual alert settings were the very thing, that got me into xDrip.

The persistent high alert is a glucose level alert. The only difference between that one and the "normal" glucose level alerts is, that the persistent one doesn't go off straight away - why shouldn't it be adjustable by a time plan just like all the other glucose level alerts?

>Before I go any further, please note that xDrip is only for research. No one should use xDrip for controlling their diabetes. A diabetic needs to see a doctor and follow the advice of their doctor including what app to use.

This might be something, xDrip has to state for legal/medical reasons, but I'm sure 99.9% of xDrip users are using it for controlling their diabetes. And why should a research tool offer any alert in the first place?

>The purpose of the persistent high alert is for someone who is not on a low-carb diet. I am not. So, it is something I depend on. When I go to sleep, if I go high, I don't want to get an alert as soon as I go high. But, if it stays high and does not come back down, after a set time period, I would like to be notified. That's what the persistent high alert does for me.

You are actually describing exactely my setting. I want to give my sugar levels a chance to come down again and I don't want to wake my partner if not really necessary. I think that is a very relatable/common topic. For daytime you and I don't need a persistent high alert and we're not in danger of forgetting about alerts, because they all are set anyway in glucose level alerts.

Isn't it just a matter of logic, to be able to turn the persistent high off at day, the same way we (would) turn off the normal high alert at night by the help of the time plan settings?

**Reply by Navid200** [reply]
I don't want to add time of day because I don't believe it is a good idea.  
Someone could forget that they have a different setting during the day and do something and find out later that because they forgot, xDrip did not alert them.  

xDrip is not consistent because it is open source.  There are many aspects in xDrip that I would not have done the way it is today.
The fact that the glucose level alerts have time of day is not a reason for me to add the same to persistent high alert.

If you never need a persistent high alert during the day because you are aware, then it will never alert you.  So, you won't need to disable it during the day.

**Comment by Teute0815** [comment]
Okay, I will let got of it (again)  ;-)
I will just add one last argument here (for now)  ;-)

I'm still a bit puzzled by your rejection, especially considering you have requested that feature (implicitly) yourself #1762 (a long time ago, before I did)
> Please enhance the other alerts (calibration, missed reading, fast rising, fast falling, persistent high, and forecast low) to offer the missing features described above.

I'm a fan of having consistent alert settings as well.

And it could also be a chance to do clean-up of settings which seems to be one of your other great goals for this project.
Instead of having a somehow hidden extra setting/menu for persistent alarms with several adjustment options, which have (had) to be incorporated one by one, it could be part of the standard "glucose level alerts" as one extra feature there (between "Threshold" and "Default Snooze" could sit a checkbox "persistent" + "for _____ minutes" for example). Followed by all the other settings that are already there. BTW, "Alert Volume Profile" could/should be incorporated in "Glucose Level alerts" as well. That way it could lose it's own menu and would make it more adjustable to personal preference (have a normal Low/High or a persistent alert ascending, have a Very Low/Very High alert blast at you straigt away.

Please don't get me wrong - I really want to thank you for your input. And I totally admire the work you guys put into this project and I'm absolutely fine, if something is just not applicable or not wanted for good reasons. On this topic I just don't agree - having seen other people requesting it as well and wondering, why you - to my understanding - reject your own requests/contradict your aim here.

**Reply by Navid200** [reply]
The persistent high alert does not have the same objective as a glucose level alert.  That's why it has its own menu.  It is the same analogy as the forecast low alert.  One may suggest that we should just make the forecast low alert to be an option on the low glucose level alert.  For the same reason, that would also not be a good idea.

The forecast low alert complements the low glucose levels.
The persistent high alert complements the high glucose level alerts.

Give me a chance to add the ability of using volume profile to the persistent high alert.  Then, try the ascending volume profile with it.  If you still think at that point that you need time of day, we can talk about that.

There are other developers.  They can add time of day to persistent high alert.  In my opinion that will be a wrong move.  But, I have no control over what happens and what does not.
All I can do is to take small steps to improve xDrip when I can and hope for avoiding to mess anything up in the process.  This is not easy work.  It is challenging.  And I am grateful that I have been given the opportunity to be part of this project.
I am not the boss and what I say does not always happen.
